layers/shared_classes_layer/python/data_service.py

```diff
@@ -69,3 +69,65 @@ def delete_marker(self, markerId):
         except Exception as e:
             logger.error(f"Error deleting marker with markerId {markerId}: {e}")
             return None
+
+    def update_marker(self, marker: LocationMarker) -> LocationMarker:
+        """
+        Update an existing marker in DynamoDB
+        :param marker: A LocationMarker with updated location details. Replaces marker with identical ID
+        :return: The updated LocationMarker.
+        :raises Exception: Raises an exception if there is an issue updating the marker.
+        """
+        try:
+            marker_id = marker.get_marker_id()
+            if not marker_id:
+                raise ValueError("Marker must have an ID")
+
+            #get original marker data
+            original_marker_data = self.table.get_item(
+                Key={'markerId': str(marker_id)}
+            ).get('Item')
+
+            if not original_marker_data:
+                raise ValueError(f"Marker with ID {marker_id} does not exist")
+
+            #copy the original marker created date
+            original_date_created = original_marker_data['dateCreated']
+            marker.set_date_created(datetime.fromisoformat(original_date_created))
```

Comment:
@PCBush the copying of original date is redundant. Please remove it

Reply:
Removed